cnxepub/adapters.py

```diff
@@ -437,7 +440,7 @@ def _compute_id(p, elem, key):
             except ValueError:
                 pass
         else:  # Punt - no parent uuid, make one up for child
-            return str(uuid.uuid4())
+            raise ValueError('Could not find parent UUID while trying to generate a UUID', p_ids)
```

Comment:
My hunch is that this is safe if we can sort out the above, so we can avoid regression with top level pages.